Code/src/main/java/nl/utwente/group10/haskell/type/ConstT.java

```diff
@@ -71,28 +61,30 @@ public String toHaskellType(final int fixity) {
     }
 
     @Override
-    public ConstT getFresh() {
-        return new ConstT(this.constructor, this.getFreshArgs());
+    protected ConstT getFreshInstance(IdentityHashMap<TypeVar.TypeInstance, TypeVar> staleToFresh) {
+    	return new ConstT(this.constructor, this.getFreshArgs(staleToFresh));
     }
 
-    /**
-     * Returns an array of fresh arguments. Selectively calls {@code getFresh} on each argument. When the same Type
-     * instance appears multiple times in the arguments no new type is instantiated. Instead, the fresh type is reused.
-     *
+
+	@Override
+	public boolean containsOccurenceOf(TypeVar tvar) {
+		for (Type t : this.args) {
+			if (t.containsOccurenceOf(tvar)) {
+				return true;
+			}
+		}
+			
+		return false;
+	}
```

Comment:
Whitespace seems to bit off (here and elsewhere) - indents are four spaces, not tabs.